src/shared/components/RichTable.tsx

```diff
@@ -0,0 +1,666 @@
+import "./RichTable.scss";
+
+import React, {
+  useState,
+  useMemo,
+  useEffect,
+  useRef,
+  useCallback
+} from "react";
+import {
+  FormControl,
+  FormCheck,
+  Table,
+  Row,
+  Col,
+  Button
+} from "react-bootstrap";
+import nanoid from "nanoid";
+import axios, { Method } from "axios";
+
+import { useApiGet } from "../../api/hooks";
+import { APIError, Pagination } from "../types";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+/**
+ * RichTable describes a component that displays tabular data in an interactive manner. It defines two main exports:
+ *
+ * - useRichTable, a React hook for creating rich tables.
+ * - RichTable, a React component to effortlessly insert a rich table.
+ *
+ * The interface is inspired by react-table.
+ */
+
+/**
+ * First, we must introduce some basic types.
+ */
+
+export type Accessor<D extends object> = (row: D) => any;
+
+export interface ColumnProps<D extends object = {}> {
+  Header: string;
```

Comment:
Any particular reason this is uppercase?

Reply:
It's a holdover from react-table, which follows the same convention. I was halfway through testing that out when I decided to roll this, hence the capitalization.

Reply:
Can it be renamed? Usually we capitalize things because they're going to be JSX, which is misleading especially for a header, which could be either JSX or a string.

Reply:
Done: 70010b3